term-background.bash

```diff
@@ -104,107 +115,146 @@ exit_if_not_sourced() {
 #
 # User should set up RGB_fg and RGB_bg arrays
 xterm_compatible_fg_bg() {
-    typeset fg bg junk
-    stty -echo 2>/dev/null
-    # Issue command to get both foreground and
-    # background color
-    #            fg       bg
-    echo -ne '\e]10;?\a\e]11;?\a'
-    IFS=: read -t 0.1 -d $'\a' x fg
-    IFS=: read -t 0.1 -d $'\a' x bg
-    stty echo 2>/dev/null
-    [[ -z $bg ]] && return 1
-    typeset -p fg
-    typeset -p bg
-    IFS='/' read -ra RGB_fg <<< $fg
-    IFS='/' read -ra RGB_bg <<< $bg
-    typeset -p RGB_fg
-    typeset -p RGB_bg
-    return 0
+  typeset fg bg
+  # Turn TTY off
+  stty -echo 2>/dev/null
+  # Issue command to get foreground
+  echo -ne '\e]10;?\a'
+  # Read back in terminal program reporting its OSC fg values
+  IFS=: read -t 0.1 -d $'\a' x fg
+  # Turn TTY back on
+  stty echo 2>/dev/null
+  # Remove any escape or control characters
+  # Note: gnome-terminal tacked \e at end
+  fg=$(echo "$fg" | sed 's/[^a-zA-Z0-9/]//g')
+  [[ -z $fg ]] && return 1
+  # Convert to array
+  typeset -p fg
+  IFS='/' read -ra RGB_fg <<<"$fg"
+  typeset -p RGB_fg
+
+  # Turn TTY off
+  stty -echo 2>/dev/null
+  # Issue command to get background
+  echo -ne '\e]11;?\a'
+  # Read back in terminal program reporting its OSC bg values
+  # purposely reading x, then discarding it.
+  IFS=: read -t 0.1 -d $'\a' x bg
+  # Turn TTY back on
+  stty echo 2>/dev/null
+  # Remove any escape or control characters
+  # Note: gnome-terminal tacked \e at end
+  bg=$(echo "$bg" | sed 's/[^a-zA-Z0-9/]//g')
+  [[ -z $bg ]] && return 1
+  # Convert to array
+  typeset -p bg
+  IFS='/' read -ra RGB_bg <<<"$bg"
+  typeset -p RGB_bg
+  xterm_osc_done=1
+  return 0
 }
 
 # From a comment left duthen in my StackOverflow answer cited above.
 osx_get_terminal_fg_bg() {
-    if [[ -n $COLORFGBG ]] ; then
-	method="COLORFGBG"
-	is_dark_colorfgbg
-    else
-	RGB_bg=($(osascript -e 'tell application "Terminal" to get the background color of the current settings of the selected tab of front window'))
-	# typeset -p RGB_bg
-	(($? != 0)) && return 1
-	TERMINAL_COLOR_MIDPOINT=117963
-	is_dark_rgb ${RGB_bg[@]}
-	method="OSX osascript"
-	success=1
-    fi
+  if [[ -n $COLORFGBG ]]; then
+    method="COLORFGBG"
+    is_dark_colorfgbg
+  else
+    RGB_bg=($(osascript -e 'tell application "Terminal" to get the background color of the current settings of the selected tab of front window'))
+    retsts=$?
+    # typeset -p RGB_bg
+    ((retsts != 0)) && return 1
+    is_dark_rgb ${RGB_fg[@]} ${RGB_bg[@]}
+    method="OSX osascript"
+    success=1
+  fi
 }
 
-
 typeset -i success=0
 typeset -i is_dark_bg=0
 typeset -i exitrc=0
+typeset -i xterm_osc_done=0
 
-get_default_bg
-
-if [[ ${BASH_VERSINFO[5]} == *"darwin"* ]] ; then
-    osx_get_terminal_fg_bg
+# Pre-analysis for non-COLORFGBG terminals
+if [ 3711 -lt "$VTE_VERSION" ] && [ -z "$COLORFGBG" ]; then
```

Comment:
since this is bash  and more broadly POSIX shell, I think  `((3711 < VTE_VERSION ))` is the more idomatic, and faster and nicer looking for the part before the `&&`

Reply:
They are in integer format, we're not comparing strings, I think.

Reply:
I will defer to you for suggested action.
It is just that JetBrain syntax analyzer pointed this out and suggested a correction. We can go back. It's up to you.